[SecuritySolution][Endpoint][ResponseActions] Response actions telemetry tests for third party agents

ashokaditya opened the pull request:
## Summary

This is a follow up of https://github.com/elastic/kibana/pull/221518. 

- [x] Adds missing tests for action creation and action response events for third party agents.
- [x] Adds telemetry events for completed action responses for crowdstrike agents.

**Note:** The action creation tests in each agent type is redundant as the action creation telemetry event is sent from a method in the base action client class, but tests are added here just to ensure that telemetry events are not broken by any changes in the future for third party agents.

### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the cloud and added to the [docker list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking changes have been approved by the breaking-change committee. The `release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed
- [ ] The PR  description includes the appropriate Release Notes section, and the correct `release_note:*` label is applied per the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [ ] Review the [backport guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing) and apply applicable `backport:*` labels.

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...





<!--ONMERGE {"backportTargets":["8.19","9.1"]} ONMERGE-->

ashokaditya committed:
telemetry tests for MS defender for endpoint

refs:
https://github.com/elastic/kibana/pull/221518
https://github.com/elastic/security-team/issues/7466

ashokaditya committed:
update sentinel one telemetry tests

https://github.com/elastic/kibana/pull/221518
https://github.com/elastic/security-team/issues/7466

ashokaditya committed:
update crowdstrike telemetry tests

refs:
https://github.com/elastic/kibana/pull/221518
https://github.com/elastic/security-team/issues/7466

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/314236)
* Commit: 28cba3912cec580860442f04acfd0ee4f54c877b

### Failed CI Steps
* [Osquery Cypress Tests on Serverless #8](https://buildkite.com/elastic/kibana-pull-request/builds/314236#0197c24f-9ec9-4a29-b66c-1c032016ad9c)

cc @ashokaditya

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/314236","number":314236,"commit":"28cba3912cec580860442f04acfd0ee4f54c877b"}],"number":314236}
buildkite-pr-comment-->

tomsonpl reviewed (approved):
Lgtm 👍

ashokaditya committed:
crowdstrike runscript tests

also includes telemetry events tests

ashokaditya committed:
add telemetry event for CS response actions

refs:
https://github.com/elastic/kibana/pull/221518
https://github.com/elastic/security-team/issues/7466

ashokaditya committed:
update tests for response telemetry events

refs:
https://github.com/elastic/kibana/pull/221518
https://github.com/elastic/security-team/issues/7466

ashokaditya committed:
Merge branch 'main' into task/edr-response-actions-telemetry-tests-ms-defender-endpoint

elasticmachine commented:
Pinging @elastic/security-defend-workflows (Team:Defend Workflows)

ashokaditya commented on x-pack/solutions/security/plugins/security_solution/server/endpoint/services/actions/clients/crowdstrike/crowdstrike_actions_client.test.ts:
```diff
@@ -328,6 +418,279 @@ describe('CrowdstrikeActionsClient class', () => {
+      });
+
+      // FIXME: What is the correct way to create a failed CS action response
+      it.skip('should send `runscript` action response telemetry event for failed action', async () => {
```
@tomsonpl do you know how to test this correctly for a failed action response?

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/314617)
* Commit: d4e9834daefabf140daecc1d26f44a73c08804f9

### Failed CI Steps
* [Osquery Cypress Tests on Serverless #8](https://buildkite.com/elastic/kibana-pull-request/builds/314617#0197c6a9-79ba-483b-828d-393af771dfde)

### History
* :broken_heart: [Build #314316](https://buildkite.com/elastic/kibana-pull-request/builds/314316) failed 28cba3912cec580860442f04acfd0ee4f54c877b
* :broken_heart: [Build #314236](https://buildkite.com/elastic/kibana-pull-request/builds/314236) failed 28cba3912cec580860442f04acfd0ee4f54c877b

cc @ashokaditya

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/314617","number":314617,"commit":"d4e9834daefabf140daecc1d26f44a73c08804f9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/314316","number":314316,"commit":"28cba3912cec580860442f04acfd0ee4f54c877b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/314236","number":314236,"commit":"28cba3912cec580860442f04acfd0ee4f54c877b"}],"number":314617}
buildkite-pr-comment-->

ashokaditya committed:
Merge branch 'main' into task/edr-response-actions-telemetry-tests-ms-defender-endpoint

ashokaditya committed:
wip logs for testing telemetry on CS

undo before merge

ashokaditya committed:
fix edge case error when no ids in search results

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/315317)
* Commit: 923e09fbdd9208ad1b3f4fa1a7c5390422266429

### Failed CI Steps
* [Linting](https://buildkite.com/elastic/kibana-pull-request/builds/315317#0197cf97-5603-48f0-93aa-e8f5324ddcad)

### History
* :yellow_heart: [Build #314834](https://buildkite.com/elastic/kibana-pull-request/builds/314834) was flaky 17b8efb7c1a86d0b52d3cdc51fbbea613c910f5b
* :broken_heart: [Build #314617](https://buildkite.com/elastic/kibana-pull-request/builds/314617) failed d4e9834daefabf140daecc1d26f44a73c08804f9
* :broken_heart: [Build #314316](https://buildkite.com/elastic/kibana-pull-request/builds/314316) failed 28cba3912cec580860442f04acfd0ee4f54c877b
* :broken_heart: [Build #314236](https://buildkite.com/elastic/kibana-pull-request/builds/314236) failed 28cba3912cec580860442f04acfd0ee4f54c877b

cc @ashokaditya

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/315317","number":315317,"commit":"923e09fbdd9208ad1b3f4fa1a7c5390422266429"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/314834","number":314834,"commit":"17b8efb7c1a86d0b52d3cdc51fbbea613c910f5b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/314617","number":314617,"commit":"d4e9834daefabf140daecc1d26f44a73c08804f9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/314316","number":314316,"commit":"28cba3912cec580860442f04acfd0ee4f54c877b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/314236","number":314236,"commit":"28cba3912cec580860442f04acfd0ee4f54c877b"}],"number":315317}
buildkite-pr-comment-->

ashokaditya committed:
Revert "fix edge case error when no ids in search results"

This reverts commit 923e09fbdd9208ad1b3f4fa1a7c5390422266429.

ashokaditya committed:
Revert "wip logs for testing telemetry on CS"

This reverts commit abf872c21f3c9e275f62cdcb37b32069a87e7185.

ashokaditya committed:
fix test mock

tomsonpl reviewed (approved):
LGTM 👍

kibanamachine commented:
Starting backport for target branches: 8.19, 9.1

https://github.com/elastic/kibana/actions/runs/16053554257

kibanamachine commented:
Starting backport for target branches: 8.19, 9.1

https://github.com/elastic/kibana/actions/runs/16053560683

kibanamachine commented:
## 💔 Some backports could not be created

| Status | Branch | Result |
|:------:|:------:|:------|
|❌|8.19|Backport failed because of merge conflicts|
|✅|9.1|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/226446">](https://github.com/elastic/kibana/pull/226446)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Manual backport
To create the backport manually run:
```
node scripts/backport --pr 225916
```

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 9.6.6 -->
